### DISCUSSION TITLE
Remove clsx and tailwind-merge (as they are already part of tailwind-variants)

### DISCUSSION THREAD

**Discussion by sohamnandi77** [irsyadadl/intentui#271] [Ideas]
The project currently includes both clsx and tailwind-merge as dependencies. However, the tailwind-variants package already includes the functionality of clsx, tailwind-merge making the use of both redundant. Can we somehow remove it and some on bundle size ?

**Comment by irsyadadl** [comment]
Do you test it without using one of them and just rely on tailwind-variant?

**Comment by sohamnandi77** [comment]
I tried for clsx , it can be removed for sure. Regarding `tailwind-merge` i am not sure, but it seems to be already a dependency of `tailwind-merge` so i believe we can remove that as well. 

`cnBase` of tailwind-variants has very similar implementation to `clsx`

https://github.com/nextui-org/tailwind-variants/blob/e312fef17537357c40726853db3f61320de0a09f/src/index.js#L21C1-L22C1


```tsx
import type { ClassValue } from "tailwind-variants";
import  { cnBase as cx } from "tailwind-variants";
import { twMerge } from "tailwind-merge";


export const cn = (...inputs: ClassValue[]): string => twMerge(cx(...inputs));

export function composeTailwindRenderProps<T>(
  tw: string | (string | undefined)[],
  className: string | ((v: T) => string) | undefined = "",
): string | ((v: T) => string) {
  return composeRenderProps(className, (className) => twMerge(tw, className));
}
```

**Comment by irsyadadl** [comment]
Interesting, I'll try it first so we don't have any problems with this.

**Comment by irsyadadl** [comment]
Struggle with React Aria components? The `ClassValue` from clsx works differently compared to Tailwind's variants. 

Edited: I think I'll sleep on it. Thanks for remembering me @sohamnandi77

**Comment by sohamnandi77** [comment]
oh okay, yeah i guess, `ClassValue` in `tailwind-variants` doesn't support dictionarys'. Sorry, i didn't tried it first, before raising a issue.
